lib/content_publisher/featured_document_migrator.rb

```diff
@@ -0,0 +1,33 @@
+module ContentPublisher
+  class FeaturedDocumentMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
+
+      public_url = Whitehall.url_maker.public_document_url(edition)
+      public_updated_at = (edition.public_timestamp || edition.updated_at)
+
+      features = Feature.includes(:feature_list).where(document_id: document.id)
+
+      features.each do |feature|
```

Comment:
And if you're editing this you might find this is nicer as `Feature.includes(:feature_list).where(document_id: document.id).each` to save a line - but no biggie.